packages/api-request-builder/src/default-params.js

```diff
@@ -62,23 +63,104 @@ export function getDefaultSearchParams (): ServiceBuilderDefaultParams {
   }
 
-  if (this.features.indexOf(features.projection) >= 0)
+  if (this.features.includes(features.projection))
     this.params.staged = true
 
-  if (this.features.indexOf(features.suggest) >= 0)
+  if (this.features.includes(features.suggest))
     this.params.searchKeywords = []
 }
+
+const hasKey = (obj: {}, key: string): boolean =>
+  Object.prototype.hasOwnProperty.call(obj, key)
+
+/**
+ * Set the supplied parameters given the current service object.
+ *
+ * @return {void}
+ */
+export function setParams (params: ServiceBuilderParams) {
+  // verify params
+  const knownKeys = [
+    'expand', 'id', 'key', 'customerId', 'sort', 'page', 'perPage', 'staged',
+    'priceCurrency', 'priceCountry', 'priceCustomerGroup', 'priceChannel',
+    'text', 'fuzzy', 'fuzzyLevel', 'markMatchingVariants', 'facet', 'filter',
+    'filterByQuery', 'filterByFacets', 'searchKeywords', 'where', 'version',
+  ]
+  Object.keys(params).forEach((key: string) => {
+    if (!knownKeys.includes(key)) throw new Error(`Unknown key "${key}"`)
+  })
+
+  // query-expand
+  if (params.expand)
+    params.expand.forEach((expansion: string) => { this.expand(expansion) })
+
+  // query-id
+  if (hasKey(params, 'id')) this.byId(params.id)
+  if (hasKey(params, 'key')) this.byKey(params.key)
+  if (hasKey(params, 'customerId')) this.byCustomerId(params.customerId)
+
+  // query-page
+  if (params.sort) params.sort.forEach(
+    (sortDesc: { by: string, direction: 'asc' | 'desc' }) => {
+      this.sort(sortDesc.by, sortDesc.direction === 'asc')
+    },
+  )
+  if (hasKey(params, 'page')) this.page(params.page)
+  if (hasKey(params, 'perPage')) this.perPage(params.perPage)
+
+  // query-projection
+  if (hasKey(params, 'staged')) this.staged(params.staged)
+  if (hasKey(params, 'priceCurrency')) this.priceCurrency(params.priceCurrency)
+  if (hasKey(params, 'priceCountry')) this.priceCountry(params.priceCountry)
+  if (hasKey(params, 'priceCustomerGroup'))
+    this.priceCustomerGroup(params.priceCustomerGroup)
+  if (hasKey(params, 'priceChannel')) this.priceChannel(params.priceChannel)
+
+  // query-search
+  if (params.text) this.text(params.text.value, params.text.language)
+  if (params.fuzzy) this.fuzzy() // boolean switch
+  if (hasKey(params, 'fuzzyLevel')) this.fuzzyLevel(params.fuzzyLevel)
+  if (params.markMatchingVariants) this.markMatchingVariants() // boolean switch
+  if (params.facet)
+    params.facet.forEach((facet: string) => { this.facet(facet) })
+  if (params.filter)
+    params.filter.forEach((filter: string) => { this.filter(filter) })
+  if (params.filterByQuery)
+    params.filterByQuery.forEach(
+      (query: string) => { this.filterByQuery(query) },
+    )
+  if (params.filterByFacets)
+    params.filterByFacets.forEach(
+      (facet: string) => { this.filterByFacets(facet) },
+    )
+
+  // query-suggest
+  if (params.searchKeywords)
+    params.searchKeywords.forEach(
+      (searchKeyword: {value: string, language: string }) => {
```

Comment:
So far this library has been using `lang` when referring to variables internally.

I chose `language` deliberately, as
- it's coming from the application 
- it would be  weird to have `language` shortened to `lang`, but `value` not shortened to `val`